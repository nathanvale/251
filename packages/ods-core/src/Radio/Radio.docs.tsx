/* eslint-disable react/display-name */
import React from "react";
import { ComponentDocs } from "@origin-digital/ods-types";
import { getSwitchDocs } from "../_private/components/AbstractSwitch/switch-docs";
import { Radio, RadioProps } from "./Radio";

export const docs: ComponentDocs<RadioProps> = {
  category: "Interaction",
  componentName: "Radio",
  description:
    "This is an accessible Radio component. It is uncontrolled by default. But if the `checked` value is provided it becomes controlled",
  propDescriptions: getSwitchDocs("radio"),
  migrationGuide: false,
  examples: [
    {
      label: "Uncontrolled",
      description: "Simple use case of an uncontrolled radio",
      Code: () => <Radio label="Accept Ts&Cs" id="tc-radio" />,
    },
    {
      label: "Controlled",
      description: "It becomes controlled when checked prop is provided",
      Code: () => <Radio label="Accept Ts&Cs" id="tc-radio" checked />,
    },
    {
      label: "Error",
      description: "Only the icon turns red when it is in error state",
      Code: () => <Radio label="Accept Ts&Cs" id="tc-radio" error />,
    },
    {
      label: "Disabled",
      description: "All turn grey when it is disabled",
      Code: () => <Radio label="Accept Ts&Cs" id="tc-radio" disabled checked />,
    },
    {
      label: "Showing Helper Text",
      description:
        "One could put a helper text under the label for further details",
      Code: () => (
        <Radio
          label="Accept Ts&Cs"
          id="tc-radio"
          helperText="Make sure you read our Ts&Cs carefully."
        />
      ),
    },
  ],
  snippets: [
    {
      label: "Uncontrolled",
      Code: () => <Radio label="Checkbox 1" id="radio1" />,
    },
  ],
};
