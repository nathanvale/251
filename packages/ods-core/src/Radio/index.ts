export { Radio } from "./Radio";
export type RadioProps = import("./Radio").RadioProps;
