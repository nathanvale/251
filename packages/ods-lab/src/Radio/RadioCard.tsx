import * as React from "react";
import makeStyles from "@material-ui/styles/makeStyles";
import {
  ToggleButtonGroup,
  ToggleButton,
  Grid,
  FormLabel,
  RadioBaseProps,
  RadioBase,
  RadioProps,
} from "..";
import {Divider} from "@origin-digital/ods-core";

export interface ExtendedRadioBaseProps extends RadioBaseProps {
  content?: React.ReactNode;
}

export interface RadioCardProps extends RadioProps {
  "data-id": string;
  label: string;
  options: Array<ExtendedRadioBaseProps>;
  handleChange: (e: string | number) => void;
  value: number | string;
}

const useStyles = makeStyles({
  largerToggleButtonGroup: {
    display: "flex",
    flexDirection: "row",
    flexWrap: "wrap",
    // "& .MuiDivider-root": {
    //   marginRight: -12,
    //   marginLeft: -12,
    // },
  },
  toggleButton: {
    minWidth: 184,
    minHeight: 266,
    marginRight: 24,
    marginBottom: 24,
    "&:hover": {
      "& .MuiRadio-root": {
        background: "unset",
      },
    },
    "& > .Mui-checked": {
      background: "unset",
    },
  },
});

export const RadioCard = (props: RadioCardProps) => {
  const styles = useStyles({});
  const {value, options, label, handleChange, "data-id": dataId} = props;
  const updateToggleGroup = (
    _event: React.MouseEvent<HTMLElement>,
    newValue: string,
  ) => {
    handleChange(newValue);
  };
  const updateRadio = (event: React.ChangeEvent<HTMLInputElement>) => {
    handleChange(event.target.value);
    event.stopPropagation();
  };
  return (
    <Grid container direction="column">
      <Grid item>
        <FormLabel>{label}</FormLabel>
      </Grid>
      <Grid item>
        <ToggleButtonGroup
          classes={{root: styles.largerToggleButtonGroup}}
          value={value}
          exclusive
          onChange={updateToggleGroup}
        >
          {options.map(r => (
            <ToggleButton
              classes={{root: styles.toggleButton}}
              value={r.value}
              key={r.value}
              selected={r.value === value}
            >
              <Grid container direction="column">
                <Grid item>
                  <RadioBase
                    key={`${r.value}-${r.label}`}
                    data-id={dataId}
                    value={r.value}
                    checked={r.value === value}
                    onChange={updateRadio}
                    disabled={r.disabled}
                    color={r.color ? r.color : "default"}
                    labelPlacement={r.labelPlacement}
                    label={r.label}
                    helpText={r.helpText}
                  />
                </Grid>
                <Grid item>
                  {r.content && (
                    <>
                      <Divider />
                      {r.content}
                    </>
                  )}
                </Grid>
              </Grid>
            </ToggleButton>
          ))}
        </ToggleButtonGroup>
      </Grid>
    </Grid>
  );
};
