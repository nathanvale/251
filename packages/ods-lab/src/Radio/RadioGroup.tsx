import * as React from "react";
import MuiRadioGroup, {
  RadioGroupProps as IRadioGroup,
} from "@material-ui/core/RadioGroup";

export interface RadioGroupProps extends IRadioGroup {
  "data-id"?: string;
}

export const RadioGroup = (props: RadioGroupProps) => (
  <MuiRadioGroup {...props}>{props.children}</MuiRadioGroup>
);
