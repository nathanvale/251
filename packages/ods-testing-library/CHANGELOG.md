# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.3.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-testing-library@0.2.0...@origin-digital/ods-testing-library@0.3.0) (2020-02-12)


### Features

* **tests, columns:** alignY default changed to top ([c23c964](https://bitbucket.orgn.io/od/origin-ui/commits/c23c964))





# 0.2.0 (2020-01-31)


### Features

* set up jest for snapshots ([6e6fbfa](https://bitbucket.orgn.io/od/origin-ui/commits/6e6fbfa))
