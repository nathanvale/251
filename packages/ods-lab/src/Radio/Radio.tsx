import React from "react";
import {RadioProps as IRadio} from "@material-ui/core/Radio";
import MuiRadioGroup, {RadioGroupProps} from "@material-ui/core/RadioGroup";
import {FormLabel, FormControl, RadioBase} from "..";

export interface IRadioProps extends IRadio {
  label: string;
  value: number | string;
  description?: string;
  error?: boolean;
  border?: boolean;
  helpText?: string | React.ReactNode;
  labelPlacement?: "start" | "end";
}

export interface RadioProps extends Array<IRadioProps>, RadioGroupProps {
  "data-id"?: string;
  label: string;
  radios: IRadioProps[];
}

export const Radio = (props: RadioProps) => {
  return (
    <FormControl component="fieldset" data-id={props["data-id"]}>
      <FormLabel component="legend">{props.label}</FormLabel>
      <MuiRadioGroup
        aria-label={props.label}
        onChange={props.onChange}
        value={props.value}
        defaultValue={props.defaultValue}
      >
        {props.radios.map(r => (
          <RadioBase
            error={r.error}
            key={r.value}
            data-id={props["data-id"]}
            disabled={r.disabled}
            checked={r.value === props.value}
            value={r.value}
            color={r.color}
            labelPlacement={r.labelPlacement}
            label={r.label}
            description={r.description}
            border={r.border}
            helpText={r.helpText}
          />
        ))}
      </MuiRadioGroup>
    </FormControl>
  );
};
