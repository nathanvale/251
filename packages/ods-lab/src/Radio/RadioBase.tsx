import * as React from "react";
import MuiRadio from "@material-ui/core/Radio";
import makeStyles from "@material-ui/styles/makeStyles";
import { AllColors } from "../AllColors/AllColors";
import { IRadioProps } from "./Radio";
import { FormControlLabel, FormHelperText, Typography } from "..";

export interface RadioBaseProps extends IRadioProps {
  "data-id"?: string;
  onChange?: (e: React.ChangeEvent<any>, checked: boolean) => void;
}

interface InputWrapperProps {
  children: React.ReactNode;
  border?: boolean;
  checked?: boolean;
}

const newStyles = makeStyles({
  inputWrapper: {
    border: (props: InputWrapperProps) => {
      return props.border ? `1px solid ${AllColors.grey56}` : "none";
    },
    borderColor: (props: InputWrapperProps) => {
      return props.border ? `1px solid ${AllColors.lightOrange}` : "none";
    },
    borderRadius: (props: InputWrapperProps) => {
      return props.border ? 10 : 0;
    },
    padding: (props: InputWrapperProps) => {
      return props.border ? 12 : 0;
    },
    marginBottom: (props: InputWrapperProps) => {
      return props.border ? 8 : 0;
    },
  },
  inlineWrapper: {
    display: "inline-flex",
    alignItems: "center",
    justifyContent: "space-between",
    width: "100%",
  },
});

const InputWrapper = (props: InputWrapperProps) => {
  const styles = newStyles(props);
  return <div className={styles.inputWrapper}>{props.children}</div>;
};

interface InlineWrapperProps {
  children: React.ReactNode;
}

const InlineWrapper = (props: InlineWrapperProps) => {
  const styles = newStyles(props);
  return <div className={styles.inlineWrapper}>{props.children}</div>;
};

export const RadioBase = (props: RadioBaseProps) => {
  const {
    "data-id": dataId,
    disabled,
    value,
    color,
    label,
    labelPlacement,
    helpText,
    checked,
    onChange,
    name,
    border,
    description,
    error,
  } = props;
  return (
    <InputWrapper border={border} checked={checked}>
      <InlineWrapper>
        <FormControlLabel
          key={value}
          data-id={dataId}
          disabled={disabled}
          value={value}
          checked={checked}
          onChange={onChange}
          color={color ? color : "default"}
          name={name}
          control={
            <MuiRadio
              inputProps={{ "aria-label": props.label }}
              color={color ? color : "default"}
            />
          }
          labelPlacement={labelPlacement}
          label={label}
        />
        {description && (
          <Typography variant="caption">{description}</Typography>
        )}
      </InlineWrapper>
      {helpText && <FormHelperText error={error}>{helpText}</FormHelperText>}
    </InputWrapper>
  );
};
