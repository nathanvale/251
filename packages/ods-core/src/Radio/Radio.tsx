import React from "react";
import { RadioProps as MuiRadioProps } from "@material-ui/core";
import { BaseSwitchProps } from "@origin-digital/ods-types";
import { RadioBase } from "../RadioBase/RadioBase";
import { AbstractSwitch } from "../_private/components/AbstractSwitch/AbstractSwitch";
import { SwitchMuiProps } from "../_private/components/AbstractSwitch/abstract-types";

export type RadioMuiPropsType = SwitchMuiProps<MuiRadioProps>;

export interface RadioProps extends BaseSwitchProps {
  muiProps?: RadioMuiPropsType;
}

export const Radio = (props: RadioProps) => {
  return (
    <AbstractSwitch<MuiRadioProps, RadioMuiPropsType>
      Component={RadioBase}
      {...props}
    />
  );
};

Radio.displayName = "Radio";
Radio.defaultProps = {};
