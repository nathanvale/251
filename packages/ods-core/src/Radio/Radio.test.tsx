import React from "react";
import { render } from "@origin-digital/ods-testing-library";
import { Radio } from "./Radio";

test("it renders an unchecked Radio with label", () => {
  const { container } = render(<Radio label="Check 1" id="my-radio" />);

  expect(container).toMatchSnapshot();
});

test("it shows correct label", () => {
  const { getAllByText } = render(<Radio label="Check 1" id="my-radio" />);

  const items = getAllByText("Check 1");
  expect(items).toHaveLength(1);
  expect(items[0]).toHaveClass("MuiFormControlLabel-label");
});

test("it renders a Radio in error state correctly", () => {
  const { container } = render(<Radio label="Check 1" id="my-radio" error />);

  expect(container).toMatchSnapshot();
});

test("it renders a disabled Radio", () => {
  const { container } = render(
    <Radio label="Check 1" id="my-radio" disabled />,
  );

  expect(container).toMatchSnapshot();
});
