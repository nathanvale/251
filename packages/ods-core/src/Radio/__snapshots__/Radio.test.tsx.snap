// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders a Radio in error state correctly 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}

<div>
  <div
    class="c0"
  >
    <label
      class="MuiFormControlLabel-root makeStyles-root-183"
      data-id="my-radio"
      id="my-radio-label"
    >
      <span
        aria-disabled="false"
        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-198 MuiRadio-root makeStyles-root-191 makeStyles-root-192 MuiRadio-colorSecondary MuiIconButton-colorSecondary"
        data-id="my-radio-RadioBase"
      >
        <span
          class="MuiIconButton-label"
        >
          <input
            aria-describedby="my-radio-helperText"
            class="PrivateSwitchBase-input-201"
            id="my-radio"
            type="radio"
            value=""
          />
          <div
            class="PrivateRadioButtonIcon-root-214"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-215"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </div>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        Check 1
      </span>
    </label>
    <p
      class="MuiFormHelperText-root makeStyles-inlineHelper-184"
      data-id="my-radio-helperText"
      id="my-radio-helperText"
    />
  </div>
</div>
`;

exports[`it renders a disabled Radio 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}

<div>
  <div
    class="c0"
  >
    <label
      class="MuiFormControlLabel-root makeStyles-root-274 Mui-disabled"
      data-id="my-radio"
      id="my-radio-label"
    >
      <span
        aria-disabled="true"
        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-289 MuiRadio-root makeStyles-root-282 makeStyles-root-283 MuiRadio-colorSecondary PrivateSwitchBase-disabled-291 Mui-disabled MuiIconButton-colorSecondary Mui-disabled Mui-disabled"
        data-id="my-radio-RadioBase"
        tabindex="-1"
      >
        <span
          class="MuiIconButton-label"
        >
          <input
            aria-describedby="my-radio-helperText"
            class="PrivateSwitchBase-input-292"
            disabled=""
            id="my-radio"
            type="radio"
            value=""
          />
          <div
            class="PrivateRadioButtonIcon-root-305"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-306"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </div>
        </span>
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label Mui-disabled MuiTypography-body1"
      >
        Check 1
      </span>
    </label>
    <p
      class="MuiFormHelperText-root makeStyles-inlineHelper-275 Mui-disabled"
      data-id="my-radio-helperText"
      id="my-radio-helperText"
    />
  </div>
</div>
`;

exports[`it renders an unchecked Radio with label 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}

<div>
  <div
    class="c0"
  >
    <label
      class="MuiFormControlLabel-root makeStyles-root-1"
      data-id="my-radio"
      id="my-radio-label"
    >
      <span
        aria-disabled="false"
        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-16 MuiRadio-root makeStyles-root-9 makeStyles-root-10 MuiRadio-colorSecondary MuiIconButton-colorSecondary"
        data-id="my-radio-RadioBase"
      >
        <span
          class="MuiIconButton-label"
        >
          <input
            aria-describedby="my-radio-helperText"
            class="PrivateSwitchBase-input-19"
            id="my-radio"
            type="radio"
            value=""
          />
          <div
            class="PrivateRadioButtonIcon-root-32"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-33"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </div>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        Check 1
      </span>
    </label>
    <p
      class="MuiFormHelperText-root makeStyles-inlineHelper-2"
      data-id="my-radio-helperText"
      id="my-radio-helperText"
    />
  </div>
</div>
`;
